### PR TITLE
Add a whitelist for directories the media manager is allowed to use

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Pull Request for Issue #24539 .

### Summary of Changes
This issue has already been addressed but only partially. This should cover the cases not handled by the original fix. 

As of now, it's possible to use a symlink as your files directory but symlinks nested within it are not allowed. This change adds a whitelist config to specifically allow any root paths you may be symlinking to. As long as you know where your symlinks are going (you should, right?) you will be able to have symlinked directories within the files folder. 

### Testing Instructions

Add one or more symlinks to external directories within the files folder. Go to the media manager and try to navigate to them, upload files into them, delete files from them. 

### Expected result

From the user point of view, symlink directories should indistinguishable from normal ones. 

### Actual result

Prior to this change, you can not navigate into symlink directories or (obviously) perform any tasks inside of them. 

### Documentation Changes Required

No but the current documentation doesn't seem to specify whether symlinks are supported or not. Since they're just a normal thing, I think they should be supported unless there is a specific reason no to do so.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_media.ini [hunk 1]
```diff
@@ -95,6 +95,10 @@ COM_MEDIA_THUMBNAIL_VIEW="Thumbnail View"
 COM_MEDIA_TITLE="Image Title"
 COM_MEDIA_UPLOAD_COMPLETE="Upload Complete: %s"
 COM_MEDIA_UPLOAD_FILE="Upload file"
+COM_MEDIA_FIELD_DIRECTORY_WHITELIST_LABEL="Directory Whitelist"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_media.ini]: Please sort keys in alpha order.

**Comment by okonomiyaki3000** [comment]
Ah, right. I forgot about that

File: administrator/language/en-GB/en-GB.com_media.ini (hunk 1 above)
Comment (okonomiyaki3000) [reply administrator/language/en-GB/en-GB.com_media.ini]: Several existing keys were out of order so I fixed them too. Also, I didn't change it but, VSCode thinks

```
COM_MEDIA="Media"
COM_MEDIA_ALIGN="Image Float"
```

Should be reversed.

**Comment by HLeithner** [comment]
Did you checked how it's implemented in J4, I'm also not sure if I would like to add this to J3 because it's more a new feature then a bugfix...

**Comment by okonomiyaki3000** [comment]
@HLeithner I haven't been able to get J4 to run, so no. It's definitely a bugfix because symlinks used to work but now they don't. You can call that a bug or a bc break but it's not possible to make the case that it's just normal operation and doesn't require any fix. With this patch, they can work again. Maybe it doesn't work as transparently or automatically as one could hope but the bottom line is that it restores functionality that was recently removed.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 2bbdb49791de4dc7bc67c6a6e34879021bef7ce2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25288">issues.joomla.org/tracker/joomla-cms/25288</a>.</sub>

**Comment by okonomiyaki3000** [comment]
I guess I need to update this branch but are we going to get another human tester?

**Comment by laoneo** [comment]
I second the argument from @HLeithner that this should be done against J4. I made a quick test and there are symlinks working as expected, even with uploads. As you can there also specify multiple directories I would say most of this pr is covered already. And if you still think there is something missing, then feel free to create your own adapter and publish it in the JED. So I'm closing it.

**Pull request closed without merging**